packages/ethereum/src/events/index.ts

```diff
@@ -20,85 +20,234 @@ export default class Event {
+                return refunds;
+            }
+
+            case 'all': {
+                return {
+                    swaps: await this.getSwapsWithStatus(swaps, _filter),
+                    refunds,
+                    withdraws,
+                } as any;
             }
 
             default: {
+                throw new Error(`Ivalind event type. Available event types: 'new', 'withdraw', 'refund', 'all'`);
             }
         }
     }
 
-    async _getPast(eventName: string, transform: Function, filter?: Function, currentBlock?: string | number) {
-        if (!currentBlock) {
-            currentBlock = await this.contract.provider.getBlockNumber();
-        }
+    async getSwapsWithStatus(swaps: SwapEvent[], filter: Filter) {
+        const ids = swaps.map((s: any) => s.id);
 
-        const eventFilter = {
-            address: this.config.contractAddress,
-            fromBlock: Number(currentBlock) - 15000, // TODO: constants
-            toBlock: currentBlock,
-            topics: [this.interface.events[eventName].topic],
-        };
+        const status = await this.contract.getStatus(ids);
 
-        const logs = await this.contract.provider.getLogs(eventFilter);
-
-        const result = logs.reduce<Log[]>((result: Log[], log: Log): Log[] => {
-            const event = this.interface.parseLog(log);
-            const t = transform(event.values, log.transactionHash);
-            const f = filter ? filter(t) : t;
-
-            if (f) {
-                result.push(f);
+        return swaps.map((s: any, index: number) => {
+            if (filter?.new?.sender?.toLowerCase() === s.sender.toLowerCase()) {
+                return { ...s, status: Number(status[index].toString()), isSender: true };
             }
+            return { ...s, status: Number(status[index].toString()) };
+        });
+    }
 
-            return result;
-        }, []);
+    async _getPast(_filter?: any, fromBlock?: string | number, toBlock?: string | number) {
+        const swaps: SwapEvent[] = [];
+        const withdraws: WithdrawEvent[] = [];
+        const refunds: RefundEvent[] = [];
 
-        return result;
-    }
+        const logsFilter = {
+            fromBlock: fromBlock || this.config.originBlock, // TODO: constants
+            toBlock: toBlock || 'latest',
+            address: this.config.contractAddress,
+        };
 
-    async subscribe(onMessage: Function, filter?: Function) {
-        this.contract.contract.on('NewContract', (...args: []) => {
-            const swap = TransformNewContract(args);
-            const result = filter ? filter(swap) : swap;
-            if (result) {
-                onMessage(result);
+        const logs = await this.contract.provider.getLogs(logsFilter);
+
+        logs.forEach((log: Log) => {
+            const parsed = this.interface.parseLog(log);
+
+            switch (parsed.name) {
+                case 'NewContract': {
+                    const swap = {
```

Comment:
Here we can make another function creator which will be something like:
```const swapObjState = swap => ({network: 'ETH', eventName: 'NEW_CONTRACT', ...parsed.values, inputAmount: parsed.values.inputAmount.toString(), outputAmount: parsed.values.outputAmount.toString(), expiration: parsed.values.expiration.toString()})```

Reusable on line 89 and 175
const swap = {...swapObjState(swap)}